fast64_internal/sm64/sm64_objects.py

```diff
@@ -1403,6 +1421,719 @@ def draw_act(self, obj, layout, value):
+        if not props.col_object and not props.gfx_object:
+            raise PluginError("No export object selected")
+
+    def get_export_objects(self, context, props):
+        if not props.export_all_selected:
+            return {props.col_object, props.gfx_object}
+
+        def obj_root(object, context):
+            while object.parent and object.parent in context.selected_objects:
+                if object.parent_type in {"ARMATURE", "OBJECT"}:
+                    return obj_root(object.parent, context)
+                else:
+                    return object
+            return object
+
+        root_objects = {obj_root(obj, context) for obj in context.selected_objects}
+        actor_objs = []
+        for obj in root_objects:
+            # eval this
+            if "Geo" in obj.sm64_obj_type or obj.sm64_obj_type in {"None", "Switch"}:
+                actor_objs.append(obj)
+
+        return actor_objs
+
+    # writes collision.inc.c file, collision_header.h
+    # writes include into aggregate file in export location (leveldata.c/<group>.c)
+    # writes name to header in aggregate file location (actor/level)
+    # var name is: const Collision <props.col_obj>_collision[]
+    def execute_col(self, props, context, obj):
+        try:
+            if props.export_col and props.obj_name_col:
+                bpy.ops.object.sm64_export_collision(export_obj=obj.name)
+        except Exception as e:
+            # pass on multiple export, throw on singular
+            if props.export_all_selected:
+                return
+            else:
+                raise Exception(e)
+
+    # writes model.inc.c, geo.inc.c file, geo_header.h
+    # writes include into aggregate file (leveldata.c/<group>.c & geo.c)
+    # writes name to header in aggregate file (header.h/<group>.h)
+    # writes model ID to model_ids.h, ID starts at prop and increments from there
+    # writes load to levels/scripts.c in appropriate group or in levels/lvl/script.c
+    # var name is: const GeoLayout <props.gfx_obj>_geo[]
+    def execute_gfx(self, props, context, obj, index):
+        try:
+            if props.export_gfx and props.obj_name_gfx:
+                if obj.type == "ARMATURE":
+                    bpy.ops.object.sm64_export_geolayout_armature(export_obj=obj.name)
+                else:
+                    bpy.ops.object.sm64_export_geolayout_object(export_obj=obj.name)
+                # write model ID, behavior, and level script load
+                self.export_model_id(context, props, index)
+                self.export_script_load(context, props)
+        except Exception as e:
+            # pass on multiple export, throw on singular
+            if props.export_all_selected:
+                return
+            else:
+                raise Exception(e)
+
+    def execute(self, context):
+        props = context.scene.fast64.sm64.combined_export
+        try:
+            self.verify_context(context, props)
```

Comment:
I'm personally not a huge fan of not just changing the context mode and then reverting it at the end of the operator, why should I not be able to export my geolayout if I'm in pose or edit mode of the armature, that's a bit dumb

Reply:
I am not sure on the backend if that causes an issue but as far as this PR goes I am not concerned with editing the internals of export operators much, which this restriction is carried over from

Reply:
it does not cause any significant issues no, in fact my settings pr does it for the tool operators, works great